gdpr/impl.go

```diff
@@ -229,53 +199,13 @@ func (p *permissionsImpl) allowID(bidder openrtb_ext.BidderName, consentMeta tcf
 	return false
 }
 
-// parseVendor parses the consent string and fetches the specified vendor's information from the GVL
-func (p *permissionsImpl) parseVendor(ctx context.Context, vendorID uint16, consent string) (parsedConsent api.VendorConsents, vendor api.Vendor, err error) {
-	parsedConsent, err = vendorconsent.ParseString(consent)
+// getVendor...
+func (p *permissionsImpl) getVendor(ctx context.Context, vendorID uint16, pc parsedConsent) (api.Vendor, error) {
+	vendorList, err := p.fetchVendorList(ctx, pc.specVersion, pc.listVersion)
 	if err != nil {
-		err = &ErrorMalformedConsent{
-			Consent: consent,
-			Cause:   err,
-		}
-		return
+		return nil, err
 	}
-
-	version := parsedConsent.Version()
-	if version != 2 {
-		return
-	}
-
-	policyVersion := parsedConsent.TCFPolicyVersion()
-	specVersion, err := getSpecVersion(policyVersion)
-	if err != nil {
-		err = &ErrorMalformedConsent{
-			Consent: consent,
-			Cause:   err,
-		}
-		return
-	}
-
-	vendorList, err := p.fetchVendorList(ctx, uint16(specVersion), parsedConsent.VendorListVersion())
-	if err != nil {
-		return
-	}
-
-	vendor = vendorList.Vendor(vendorID)
-	return
-}
-
-// getSpecVersion looks at the TCF policy version and determines the corresponding GVL specification
-// version that should be used to calculate legal basis
-func getSpecVersion(policyVersion uint8) (uint16, error) {
-	var specVersion uint16 = 3
-
-	if policyVersion > 4 {
-		return 0, fmt.Errorf("invalid TCF policy version %d", policyVersion)
-	}
-	if policyVersion < 4 {
-		specVersion = 2
-	}
-	return specVersion, nil
```

Comment:
This functionality was moved into `consent_parser.go` with the error case becoming part of `validateVersions`.